standard-controller: is trying to connect to router which was already killed

description:
when router is automatically scaled down from 3 to 2 replicas (in automated test bellow) then in standard-controller log are logged following errors: (because 3rd router was already killed)
```
{"log":"2018-03-20 10:49:16 INFO  AddressController:236 - Error requesting router status from qdrouterd-1825555819-kb2ht. Ignoring\n","stream":"stdout","time":"2018-03-20T10:49:16.925574751Z"}
{"log":"java.util.concurrent.ExecutionException: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /172.17.0.14:55671\n","stream":"stdout","time":"2018-03-20T10:49:16.92559971Z"}
{"log":"\u0009at java.util.concurrent.CompletableFuture.reportGet(CompletableFuture.java:357)\n","stream":"stdout","time":"2018-03-20T10:49:16.925602817Z"}
{"log":"\u0009at java.util.concurrent.CompletableFuture.get(CompletableFuture.java:1915)\n","stream":"stdout","time":"2018-03-20T10:49:16.925605496Z"}
{"log":"\u0009at io.enmasse.amqp.SyncRequestClient.request(SyncRequestClient.java:89)\n","stream":"stdout","time":"2018-03-20T10:49:16.925607512Z"}
{"log":"\u0009at io.enmasse.controller.standard.RouterStatusCollector.collectRouter(RouterStatusCollector.java:101)\n","stream":"stdout","time":"2018-03-20T10:49:16.925609535Z"}
{"log":"\u0009at io.enmasse.controller.standard.RouterStatusCollector.collect(RouterStatusCollector.java:61)\n","stream":"stdout","time":"2018-03-20T10:49:16.925618646Z"}
{"log":"\u0009at io.enmasse.controller.standard.AddressController.checkStatuses(AddressController.java:231)\n","stream":"stdout","time":"2018-03-20T10:49:16.925620943Z"}
{"log":"\u0009at io.enmasse.controller.standard.AddressController.onUpdate(AddressController.java:133)\n","stream":"stdout","time":"2018-03-20T10:49:16.92562293Z"}
{"log":"\u0009at io.enmasse.k8s.api.ResourceChecker.doWork(ResourceChecker.java:48)\n","stream":"stdout","time":"2018-03-20T10:49:16.925624834Z"}
{"log":"\u0009at io.enmasse.k8s.api.ResourceChecker.run(ResourceChecker.java:40)\n","stream":"stdout","time":"2018-03-20T10:49:16.925626745Z"}
{"log":"\u0009at java.lang.Thread.run(Thread.java:748)\n","stream":"stdout","time":"2018-03-20T10:49:16.925628574Z"}
{"log":"Caused by: io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /172.17.0.14:55671\n","stream":"stdout","time":"2018-03-20T10:49:16.925630464Z"}
{"log":"\u0009at sun.nio.ch.SocketChannelImpl.checkConnect(Native Method)\n","stream":"stdout","time":"2018-03-20T10:49:16.925632392Z"}
{"log":"\u0009at sun.nio.ch.SocketChannelImpl.finishConnect(SocketChannelImpl.java:717)\n","stream":"stdout","time":"2018-03-20T10:49:16.925634378Z"}
{"log":"\u0009at io.netty.channel.socket.nio.NioSocketChannel.doFinishConnect(NioSocketChannel.java:325)\n","stream":"stdout","time":"2018-03-20T10:49:16.92563623Z"}
{"log":"\u0009at io.netty.channel.nio.AbstractNioChannel$AbstractNioUnsafe.finishConnect(AbstractNioChannel.java:340)\n","stream":"stdout","time":"2018-03-20T10:49:16.925638189Z"}
{"log":"\u0009at io.netty.channel.nio.NioEventLoop.processSelectedKey(NioEventLoop.java:633)\n","stream":"stdout","time":"2018-03-20T10:49:16.92564047Z"}
{"log":"\u0009at io.netty.channel.nio.NioEventLoop.processSelectedKeysOptimized(NioEventLoop.java:580)\n","stream":"stdout","time":"2018-03-20T10:49:16.92564249Z"}
{"log":"\u0009at io.netty.channel.nio.NioEventLoop.processSelectedKeys(NioEventLoop.java:497)\n","stream":"stdout","time":"2018-03-20T10:49:16.925644366Z"}
{"log":"\u0009at io.netty.channel.nio.NioEventLoop.run(NioEventLoop.java:459)\n","stream":"stdout","time":"2018-03-20T10:49:16.92564626Z"}
{"log":"\u0009at io.netty.util.concurrent.SingleThreadEventExecutor$5.run(SingleThreadEventExecutor.java:886)\n","stream":"stdout","time":"2018-03-20T10:49:16.925648089Z"}
{"log":"\u0009at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)\n","stream":"stdout","time":"2018-03-20T10:49:16.925650064Z"}
{"log":"\u0009... 1 more\n","stream":"stdout","time":"2018-03-20T10:49:16.92565193Z"}
{"log":"Caused by: java.net.ConnectException: Connection refused\n","stream":"stdout","time":"2018-03-20T10:49:16.925654702Z"}
{"log":"\u0009... 11 more\n","stream":"stdout","time":"2018-03-20T10:49:16.925681138Z"}
{"log":"2018-03-20 10:49:16 WARN  LogEventLogger:22 - RouterCheckFailed (kind=AddressSpace name=standard-shared-0): io.netty.channel.AbstractChannel$AnnotatedConnectException: Connection refused: /172.17.0.14:55671\n","stream":"stdout","time":"2018-03-20T10:49:16.926078645Z"}
```
This issue has no impact to functionality. Thank you @lulf  for collaboration on identification this issue!

automated test:
`make MULTITENANT=true SYSTEMTEST_ARGS='standard.AnycastTest#testScaleRouterAutomatically' systemtests`

--- a/standard-controller/src/main/java/io/enmasse/controller/standard/AddressController.java
+++ b/standard-controller/src/main/java/io/enmasse/controller/standard/AddressController.java
@@ -8,7 +8,7 @@ import io.enmasse.address.model.*;
 import io.enmasse.config.AnnotationKeys;
 import io.enmasse.k8s.api.*;
 import io.fabric8.kubernetes.api.model.Pod;
-import io.fabric8.kubernetes.api.model.extensions.Deployment;
+import io.fabric8.kubernetes.client.internal.readiness.Readiness;
 import io.vertx.core.AbstractVerticle;
 import io.vertx.core.Future;
 import org.slf4j.Logger;
@@ -226,7 +226,7 @@ public class AddressController extends AbstractVerticle implements Watcher<Addre
         RouterStatusCollector routerStatusCollector = new RouterStatusCollector(vertx, certDir);
         List<RouterStatus> routerStatusList = new ArrayList<>();
         for (Pod router : kubernetes.listRouters()) {
-            if (router.getStatus().getPodIP() != null && !"".equals(router.getStatus().getPodIP())) {
+            if (Readiness.isPodReady(router)) {
                 try {
                     RouterStatus routerStatus = routerStatusCollector.collect(router);
                     if (routerStatus != null) {

